client/components/webpack-build-monitor/index.tsx

```diff
@@ -0,0 +1,222 @@
+				setIsCssConnected( false );
+			} );
+
+			socket.on(
+				'css-hot-reload',
+				( { status }: { status: 'building' | 'build-failed' | 'reload' | 'up-to-date' } ) => {
+					debug( 'Hot CSS status update: %o', status );
+					switch ( status ) {
+						case 'building':
+							setCssBuilding( true );
+							break;
+
+						case 'build-failed':
+							setHasCssErrors( true );
+							setCssBuilding( false );
+							break;
+
+						case 'reload':
+						case 'up-to-date':
+							setHasCssErrors( false );
+							setCssBuilding( false );
+							break;
+					}
+				}
+			);
+		};
+
+		connect();
+
+		return () => {
+			if ( socket && ! socket.disconnected ) {
+				socket.close();
+			}
+		};
+	}, [] );
+
+	useEffect( () => {
+		if ( typeof EventSource === 'undefined' ) {
+			if ( process.env.NODE_ENV !== 'production' ) {
+				// eslint-disable-next-line no-console
+				console.warn( 'Webpack build monitor disabled. No `EventSource`.' );
+			}
+		} else {
+			let lastActivity: TimestampMS;
+			let connectionTimer: IntervalHandle;
+			let source: EventSource;
+			const connect = () => {
+				debug( 'Webpack HMR connecting' );
+				source = new EventSource( '/__webpack_hmr' );
+
+				source.onopen = () => {
+					debug( 'Webpack HMR connected' );
+					setIsWebpackConnected( true );
+					lastActivity = Date.now();
+					connectionTimer = setInterval( () => {
+						if ( Date.now() - lastActivity > CONNECTION_TIMEOUT ) {
+							debug( 'Webpack HMR connection timeout. Reconnecting in %o…', CONNECTION_TIMEOUT );
+							setIsWebpackConnected( false );
+							clearInterval( connectionTimer );
+							source.close();
+							setTimeout( connect, CONNECTION_TIMEOUT );
+						}
+					}, CONNECTION_TIMEOUT / 2 );
+				};
+
+				source.onmessage = ( m: MessageEvent ) => {
+					lastActivity = Date.now();
+					if ( m.data !== '💓' ) {
+						let action: string | undefined;
+						let nextErrors;
+						let nextWarnings;
+						let nextHash;
+
+						try {
+							const parsedData = JSON.parse( m.data );
+							debug( 'Webpack HMR message: %o', parsedData );
+							action = parsedData.action;
+							nextErrors = parsedData.errors;
+							nextWarnings = parsedData.warnings;
+							nextHash = parsedData.hash;
+						} catch ( err ) {
+							debug( 'Could not parse HMR message.data %o', m.data );
+						}
+
+						switch ( action ) {
+							case 'building':
+								setWebpackBuilding( true );
+								setHasWebpackErrors( false );
+								setHasWebpackWarnings( false );
+								break;
+
+							case 'built':
+								setWebpackBuilding( false );
+							// fall through
+							case 'sync':
+								setHasWebpackErrors( !! nextErrors.length );
+								setHasWebpackWarnings( !! nextWarnings.length );
+								setLastHash( nextHash );
+								break;
+						}
+					}
+				};
+			};
+
+			connect();
+
+			return () => {
+				if ( source ) {
+					source.close();
+				}
+				if ( connectionTimer ) {
```

Comment:
No need for this check; `clearInterval` is a noop on anything but a valid handle.

Reply:
Dropped it: 800213a29f116d33e182140553a810588bef2a6d